src/libAtomVM/term.h

```diff
@@ -524,14 +524,14 @@ static inline term term_from_local_process_id(uint32_t local_process_id)
 static inline term term_from_literal_binary(void *data, uint32_t size, Context *ctx)
 {
 #if TERM_BYTES == 4
-    int size_in_terms = ((size + 4 - 1) >> 2);
+    int size_in_terms = ((size + 4 - 1) >> 2) + 1;
 #elif TERM_BYTES == 8
-    int size_in_terms = ((size + 8 - 1) >> 3);
+    int size_in_terms = ((size + 8 - 1) >> 3) + 1;
 #else
     #error
 #endif
 
-    term *boxed_value = memory_heap_alloc(ctx, size_in_terms + 2);
```

Comment:
I don't really get the meaning of this change.

Reply:
Looking it it more, I am confused as well.

The number of words in the heap is still the same.  However, the size_in_terms (which gets written to the second word in the boxed term) is not properly computed.

Perhaps the division by 4 is wrong.  Suppose TERM_BYTES is 4, and size is 1.  Then 1 + 4 - 1 = 4, and 4 >> 2 is 0.  But the size in terms should be at least 1.

Let me add a test to illustrate.